src/kaleidoscope/driver/hid/Keyboardio.h

```diff
@@ -16,6 +16,9 @@
  */
 
 #pragma once
+
+#ifndef USBD_USE_HID_COMPOSITE
```

Comment:
Please forgive me if I'm being naïve, but this strikes me as something that ought to be handled by not including this header file if `USBD_USE_HID_COMPOSITE` is defined, rather than excluding it here.  I have a lot of difficulty navigating the maze of headers used by the hardware modules, so I suppose you've got a good reason for doing it this way, but I think a comment explaining why this is here in the code is warranted.